helix-rest/src/main/java/org/apache/helix/rest/server/HelixRestObjectNameFactory.java

```diff
@@ -0,0 +1,90 @@
+package org.apache.helix.rest.server;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.util.Hashtable;
+import javax.management.MalformedObjectNameException;
+import javax.management.ObjectName;
+
+import com.codahale.metrics.jmx.ObjectNameFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Creates an {@link ObjectName} that has "name", "type" and "namespace" properties
+ * for metrics registry in Helix rest service.
+ *
+ * <p>It is recommended to only be used within Helix REST.
+ */
+class HelixRestObjectNameFactory implements ObjectNameFactory {
+  private static final Logger LOG = LoggerFactory.getLogger(HelixRestObjectNameFactory.class);
+
+  private static final String KEY_NAME = "name";
+  private static final String KEY_TYPE = "type";
+  private static final String KEY_NAMESPACE = "namespace";
+
+  private final String _namespace;
+
+  HelixRestObjectNameFactory(String nameSpace) {
+    _namespace = nameSpace;
+  }
+
+  public ObjectName createName(String type, String domain, String name) {
```

Comment:
Do we allow null string input or not?

Reply:
No. `Hashtable` has the null check. If any property is null, NPE will be thrown.